operator/cmd/root.go

```diff
@@ -100,6 +101,9 @@ var (
 		"operator-controlplane",
 		"Operator Control Plane",
 
+		cell.Config(cmtypes.DefaultClusterInfo),
+		cell.Invoke(func(cinfo cmtypes.ClusterInfo) error { return cinfo.Validate() }),
```

Comment:
I think you don't need to wrap it, but rather you can just do: `cell.Invoke(cmtypes.ClusterInfo.Validate)`. Debatable whether this is less or more confusing, but it is shorter.. :D

Why do we need `Validate` here, but not in `daemon/cmd/cells.go`? If it's missing from there, how about doing `cmtypes.ConfigCell` that's `cell.Group(cell.Config(DefaultClusterInfo), cell.Invoke(ClusterInfo.Validate))`? Preferably this config would be part of some module and not referenced naked here like this, but perhaps that's to come in the future?

Reply:
> I think you don't need to wrap it, but rather you can just do: `cell.Invoke(cmtypes.ClusterInfo.Validate)`. Debatable whether this is less or more confusing, but it is shorter.. :D

Uh, that's right. I didn't realize it.

> Why do we need Validate here, but not in daemon/cmd/cells.go? If it's missing from there, how about doing cmtypes.ConfigCell that's cell.Group(cell.Config(DefaultClusterInfo), cell.Invoke(ClusterInfo.Validate))? Preferably this config would be part of some module and not referenced naked here like this, but perhaps that's to come in the future?

So, the thing is that depending on the component we need to perform a slightly different validation, mainly because for kvstoremesh we don't allow the cluster ID to be zero (which also means that the check needs to be performed from a start hook, as it is the default value). The validation in the agent is currently still part of `Config.Validate` mainly because that is performed earlier to prevent possible issues for components which do not directly depend on the `ClusterInfo` struct.